src/recipes.jl

```diff
@@ -0,0 +1,30 @@
+using RecipesBase
+
+# threshold to plot partitions rather than data
+const PTHRESH = 10_000
+
+#-----------------------------------------------------------------# table and selection
+@recipe function f(t::AbstractIndexedTable, selection::Union{Symbol, Number, Pair}; 
+                   partition = length(t) > PTHRESH, nparts = 50, reducer = nothing)
+    if partition
+        o = (reducer == nothing) ? 
+            make_reducer(fieldtype(eltype(t), IndexedTables.colindex(t, selection))) : 
+            reducer
+        @series begin 
+            title --> selection 
+            reduce(Partition(o, nparts), t; select = selection)
+        end
+    else
+        @series begin 
+            title --> selection
+            collect(select(t, selection))
+        end
+    end
+end
+
+
+# Default OnlineStat based on data type
+make_reducer(::Type{<:Number}) = Mean()
+make_reducer(::Type{T})  where {T<:Union{AbstractString, Symbol}} = CountMap(T)
```

Comment:
I guess the other common type is `TimeType` (i.e. Date, DateTime and Time)... Show a histogram?

Reply:
Currently I think `CountMap` is the only OnlineStat that could handle a `TimeType`, but `CountMap` grows in size for every unique value.  If a user knows they have a small number of dates in their data, they can use 

```julia
plot(data, date_column; reducer = CountMap(Date))
```

(after I make a small change to OnlineStatsBase to allow `TimeType` as a valid input)